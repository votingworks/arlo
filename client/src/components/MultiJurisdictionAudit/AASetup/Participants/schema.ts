import * as Yup from 'yup'

const participantsSchema = Yup.object().shape({
  state: Yup.string().required('Required'),
  csv: Yup.mixed().required('You must upload a file'),
})

export default participantsSchema
