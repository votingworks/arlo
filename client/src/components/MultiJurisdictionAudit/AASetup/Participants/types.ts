import { IAuditSettings } from '../../../../types'

export interface IValues {
  state: IAuditSettings['state']
}
