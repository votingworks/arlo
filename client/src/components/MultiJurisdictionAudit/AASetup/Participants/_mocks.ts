import { readFileSync } from 'fs'
import { join } from 'path'

const jurisdictionFile = new File(
  [
    readFileSync(
      join(
        __dirname,
        '../../../../../public/sample_jurisdiction_filesheet.csv'
      ),
      'utf8'
    ),
  ],
  'jurisdictions.csv',
  { type: 'text/csv' }
)

export default jurisdictionFile
