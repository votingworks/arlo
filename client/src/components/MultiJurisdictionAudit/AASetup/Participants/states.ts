import locale from '../../../../locales/current'

export default Object.keys(locale.usState).map(value => ({
  value,
  label: locale.usState[value as keyof typeof locale.usState],
}))
