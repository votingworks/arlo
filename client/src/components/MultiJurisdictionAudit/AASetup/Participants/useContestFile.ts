import { useEffect, useState } from 'react'
import { api } from '../../../utilities'
import { IFileInfo } from '../../useJurisdictions'

const loadContestFile = async (
  electionId: string
): Promise<IFileInfo | null> => {
  const response = await api<IFileInfo>(
    `/election/${electionId}/standardized-contests/file`
  )
  if (!response) return null
  return response
}

const putContestFileFile = async (
  electionId: string,
  csv: File
): Promise<boolean> => {
  const formData: FormData = new FormData()
  formData.append('standardize-contests', csv, csv.name)
  const response = await api(
    `/election/${electionId}/standardized-contests/file`,
    {
      method: 'PUT',
      body: formData,
    }
  )
  return !!response
}

const useContestFile = (
  electionId: string
): [IFileInfo | null, (csv: File) => Promise<boolean>] => {
  const [jurisdictionFile, setContestFile] = useState<IFileInfo | null>(null)

  const uploadContestFile = async (csv: File): Promise<boolean> => {
    // TODO poll for result of upload
    if (await putContestFileFile(electionId, csv)) {
      setContestFile(await loadContestFile(electionId))
      return true
    }
    return false
  }

  useEffect(() => {
    ;(async () => {
      setContestFile(await loadContestFile(electionId))
    })()
  }, [electionId])

  return [jurisdictionFile, uploadContestFile]
}

export default useContestFile
