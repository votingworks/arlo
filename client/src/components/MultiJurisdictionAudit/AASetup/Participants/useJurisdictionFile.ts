import { useEffect, useState } from 'react'
import { api } from '../../../utilities'
import { IFileInfo } from '../../useJurisdictions'

const loadJurisdictionFile = async (
  electionId: string
): Promise<IFileInfo | null> => {
  const response = await api<IFileInfo>(
    `/election/${electionId}/jurisdiction/file`
  )
  if (!response) return null
  return response
}

const putJurisdictionFileFile = async (
  electionId: string,
  csv: File
): Promise<boolean> => {
  const formData: FormData = new FormData()
  formData.append('jurisdictions', csv, csv.name)
  const response = await api(`/election/${electionId}/jurisdiction/file`, {
    method: 'PUT',
    body: formData,
  })
  return !!response
}

const useJurisdictionFile = (
  electionId: string
): [IFileInfo | null, (csv: File) => Promise<boolean>] => {
  const [jurisdictionFile, setJurisdictionFile] = useState<IFileInfo | null>(
    null
  )

  const uploadJurisdictionFile = async (csv: File): Promise<boolean> => {
    // TODO poll for result of upload
    if (await putJurisdictionFileFile(electionId, csv)) {
      setJurisdictionFile(await loadJurisdictionFile(electionId))
      return true
    }
    return false
  }

  useEffect(() => {
    ;(async () => {
      setJurisdictionFile(await loadJurisdictionFile(electionId))
    })()
  }, [electionId])

  return [jurisdictionFile, uploadJurisdictionFile]
}

export default useJurisdictionFile
