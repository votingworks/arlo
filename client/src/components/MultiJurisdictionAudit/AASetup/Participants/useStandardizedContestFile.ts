import { useEffect, useState } from 'react'
import { api } from '../../../utilities'
import { IFileInfo } from '../../useJurisdictions'

const loadStandardizedContestFile = async (
  electionId: string
): Promise<IFileInfo | null> => {
  const response = await api<IFileInfo>(
    `/election/${electionId}/standardized-contests/file`
  )
  if (!response) return null
  return response
}

const putStandardizedContestFile = async (
  electionId: string,
  csv: File
): Promise<boolean> => {
  const formData: FormData = new FormData()
  formData.append('standardized-contests', csv, csv.name)
  const response = await api(
    `/election/${electionId}/standardized-contests/file`,
    {
      method: 'PUT',
      body: formData,
    }
  )
  return !!response
}

const useStandardizedContestFile = (
  electionId: string
): [IFileInfo | null, (csv: File) => Promise<boolean>] => {
  const [
    jurisdictionFile,
    setStandardizedContestFile,
  ] = useState<IFileInfo | null>(null)

  const uploadStandardizedContestFile = async (csv: File): Promise<boolean> => {
    // TODO poll for result of upload
    if (await putStandardizedContestFile(electionId, csv)) {
      setStandardizedContestFile(await loadStandardizedContestFile(electionId))
      return true
    }
    return false
  }

  useEffect(() => {
    ;(async () => {
      setStandardizedContestFile(await loadStandardizedContestFile(electionId))
    })()
  }, [electionId])

  return [jurisdictionFile, uploadStandardizedContestFile]
}

export default useStandardizedContestFile
