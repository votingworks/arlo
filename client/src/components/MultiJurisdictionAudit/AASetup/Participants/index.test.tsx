import React from 'react'
import { fireEvent, waitFor, render, screen } from '@testing-library/react'
import { BrowserRouter as Router, useParams } from 'react-router-dom'
import { toast } from 'react-toastify'
import relativeStages from '../_mocks'
import Participants from './index'
import jurisdictionFile from './_mocks'
import * as utilities from '../../../utilities'
import useAuditSettings from '../../useAuditSettings'
import { fileProcessingMocks } from '../../useSetupMenuItems/_mocks'

const auditSettingsMock = useAuditSettings as jest.Mock

const apiMock: jest.SpyInstance<
  ReturnType<typeof utilities.api>,
  Parameters<typeof utilities.api>
> = jest.spyOn(utilities, 'api').mockImplementation()

const checkAndToastMock: jest.SpyInstance<
  ReturnType<typeof utilities.checkAndToast>,
  Parameters<typeof utilities.checkAndToast>
> = jest.spyOn(utilities, 'checkAndToast').mockReturnValue(false)
const toastSpy = jest.spyOn(toast, 'error').mockImplementation()

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'), // use actual for all non-hook parts
  useParams: jest.fn(),
}))
const routeMock = useParams as jest.Mock
routeMock.mockReturnValue({
  electionId: '1',
  view: 'setup',
})

jest.mock('../../useAuditSettings')
auditSettingsMock.mockReturnValue([
  {
    state: null,
    electionName: null,
    online: null,
    randomSeed: null,
    riskLimit: null,
  },
  async () => true,
])

const formData: FormData = new FormData()
formData.append('jurisdictions', jurisdictionFile, jurisdictionFile.name)

const { nextStage } = relativeStages('Participants')

const fillAndSubmit = async () => {
  render(
    <Router>
      <Participants locked={false} nextStage={nextStage} />
    </Router>
  )

  fireEvent.change(await screen.findByTestId('state-field'), {
    target: { value: 'WA' },
  })

  const csvInput = screen.getByLabelText('Select a CSV...')
  fireEvent.change(csvInput, { target: { files: [] } })
  fireEvent.blur(csvInput)
  await waitFor(() =>
    expect(screen.queryByText('You must upload a file')).toBeTruthy()
  )
  fireEvent.change(csvInput, { target: { files: [jurisdictionFile] } })
  await waitFor(() =>
    expect(screen.queryByText('You must upload a file')).toBeFalsy()
  )
  await waitFor(() =>
    expect(screen.queryByLabelText('Select a CSV...')).toBeFalsy()
  )
  await waitFor(() =>
    expect(screen.queryByLabelText('jurisdictions.csv')).toBeTruthy()
  )

  fireEvent.click(screen.getByText('Save & Next'), { bubbles: true })
}

beforeEach(() => {
  apiMock.mockClear()
  toastSpy.mockClear()
  checkAndToastMock.mockClear()
  routeMock.mockClear()
  ;(nextStage.activate as jest.Mock).mockClear()
  auditSettingsMock.mockClear()
})

describe('Audit Setup > Participants', () => {
  beforeEach(() => {
    apiMock.mockImplementation(async (endpoint: string) => {
      switch (endpoint) {
        case '/election/1/jurisdiction/file':
          return { file: null, processing: null }
        default:
          return null
      }
    })
  })

  it('renders empty state correctly', async () => {
    const { container } = render(
      <Router>
        <Participants locked={false} nextStage={nextStage} />
      </Router>
    )
    await waitFor(() => expect(apiMock).toHaveBeenCalled())
    expect(container).toMatchSnapshot()
  })

  it('renders filled state correctly & resets file', async () => {
    apiMock.mockImplementation(async (endpoint: string) => {
      switch (endpoint) {
        case '/election/1/jurisdiction/file':
          return {
            file: {
              name: 'filename.csv',
              uploadedAt: '2019-07-18T16:34:07.000Z',
            },
            processing: fileProcessingMocks,
          }
        default:
          return null
      }
    })
    const { container } = render(
      <Router>
        <Participants locked={false} nextStage={nextStage} />
      </Router>
    )
    const resetButton = await screen.findByText('Replace File')
    expect(container).toMatchSnapshot()
    fireEvent.click(resetButton, { bubbles: true })
    await screen.findByLabelText('Select a CSV...')
    expect(container).toMatchSnapshot()
  })

  it('selects a state and submits it', async () => {
    apiMock
      .mockResolvedValueOnce({ file: null, processing: null })
      .mockResolvedValue({ status: 'ok' })
    await fillAndSubmit()
    await waitFor(() => {
      expect(apiMock).toBeCalledTimes(3)
      expect(apiMock).toHaveBeenNthCalledWith(
        2,
        expect.stringMatching(/\/election\/[^/]+\/jurisdiction\/file/),
        { body: formData, method: 'PUT' }
      )
      expect(nextStage.activate).toHaveBeenCalledTimes(1)
    })
  })

  it('handles api error on /election/:electionId/jurisdiction/file', async () => {
    apiMock
      .mockResolvedValueOnce({ file: null, processing: null })
      .mockRejectedValue({ message: 'error' })

    await fillAndSubmit()

    await waitFor(() => {
      expect(apiMock).toBeCalledTimes(2)
      expect(toastSpy).toBeCalledTimes(1)
      expect(nextStage.activate).toHaveBeenCalledTimes(0)
    })
  })

  it('handles failure to update settings', async () => {
    auditSettingsMock.mockReturnValue([
      {
        state: null,
        electionName: null,
        online: null,
        randomSeed: null,
        riskLimit: null,
      },
      async () => false,
    ])

    await fillAndSubmit()

    await waitFor(() => {
      expect(apiMock).toBeCalledTimes(1)
      expect(toastSpy).toBeCalledTimes(0)
      expect(checkAndToastMock).toBeCalledTimes(0)
      expect(nextStage.activate).toHaveBeenCalledTimes(0)
    })
  })
})
