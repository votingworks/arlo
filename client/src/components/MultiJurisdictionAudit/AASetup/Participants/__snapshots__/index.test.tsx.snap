// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Audit Setup > Participants renders empty state correctly 1`] = `
<div>
  <form
    data-testid="form-one"
  >
    <div
      class="sc-bwzfXH jFbAHP"
    >
      <h2
        class="bp3-heading sc-bdVaJa ibVpPq"
      >
        Participants
      </h2>
      <div
        class="sc-gzVnrw coBAaV"
      >
        <label
          for="state"
        >
          Choose your state from the options below
          <br />
          <div
            class="bp3-html-select sc-iwsKbI iYAkEx"
          >
            <select
              data-testid="state-field"
              field="[object Object]"
              form="[object Object]"
              id="state"
            >
              <option
                value=""
              />
              <option
                label="Alabama"
                value="AL"
              >
                Alabama
              </option>
              <option
                label="Alaska"
                value="AK"
              >
                Alaska
              </option>
              <option
                label="Arizona"
                value="AZ"
              >
                Arizona
              </option>
              <option
                label="Arkansas"
                value="AR"
              >
                Arkansas
              </option>
              <option
                label="California"
                value="CA"
              >
                California
              </option>
              <option
                label="Colorado"
                value="CO"
              >
                Colorado
              </option>
              <option
                label="Connecticut"
                value="CT"
              >
                Connecticut
              </option>
              <option
                label="Delaware"
                value="DE"
              >
                Delaware
              </option>
              <option
                label="District Of Columbia"
                value="DC"
              >
                District Of Columbia
              </option>
              <option
                label="Florida"
                value="FL"
              >
                Florida
              </option>
              <option
                label="Georgia"
                value="GA"
              >
                Georgia
              </option>
              <option
                label="Hawaii"
                value="HI"
              >
                Hawaii
              </option>
              <option
                label="Idaho"
                value="ID"
              >
                Idaho
              </option>
              <option
                label="Illinois"
                value="IL"
              >
                Illinois
              </option>
              <option
                label="Indiana"
                value="IN"
              >
                Indiana
              </option>
              <option
                label="Iowa"
                value="IA"
              >
                Iowa
              </option>
              <option
                label="Kansas"
                value="KS"
              >
                Kansas
              </option>
              <option
                label="Kentucky"
                value="KY"
              >
                Kentucky
              </option>
              <option
                label="Louisiana"
                value="LA"
              >
                Louisiana
              </option>
              <option
                label="Maine"
                value="ME"
              >
                Maine
              </option>
              <option
                label="Maryland"
                value="MD"
              >
                Maryland
              </option>
              <option
                label="Massachusetts"
                value="MA"
              >
                Massachusetts
              </option>
              <option
                label="Michigan"
                value="MI"
              >
                Michigan
              </option>
              <option
                label="Minnesota"
                value="MN"
              >
                Minnesota
              </option>
              <option
                label="Mississippi"
                value="MS"
              >
                Mississippi
              </option>
              <option
                label="Missouri"
                value="MO"
              >
                Missouri
              </option>
              <option
                label="Montana"
                value="MT"
              >
                Montana
              </option>
              <option
                label="Nebraska"
                value="NE"
              >
                Nebraska
              </option>
              <option
                label="Nevada"
                value="NV"
              >
                Nevada
              </option>
              <option
                label="New Hampshire"
                value="NH"
              >
                New Hampshire
              </option>
              <option
                label="New Jersey"
                value="NJ"
              >
                New Jersey
              </option>
              <option
                label="New Mexico"
                value="NM"
              >
                New Mexico
              </option>
              <option
                label="New York"
                value="NY"
              >
                New York
              </option>
              <option
                label="North Carolina"
                value="NC"
              >
                North Carolina
              </option>
              <option
                label="North Dakota"
                value="ND"
              >
                North Dakota
              </option>
              <option
                label="Ohio"
                value="OH"
              >
                Ohio
              </option>
              <option
                label="Oklahoma"
                value="OK"
              >
                Oklahoma
              </option>
              <option
                label="Oregon"
                value="OR"
              >
                Oregon
              </option>
              <option
                label="Pennsylvania"
                value="PA"
              >
                Pennsylvania
              </option>
              <option
                label="Rhode Island"
                value="RI"
              >
                Rhode Island
              </option>
              <option
                label="South Carolina"
                value="SC"
              >
                South Carolina
              </option>
              <option
                label="South Dakota"
                value="SD"
              >
                South Dakota
              </option>
              <option
                label="Tennessee"
                value="TN"
              >
                Tennessee
              </option>
              <option
                label="Texas"
                value="TX"
              >
                Texas
              </option>
              <option
                label="Utah"
                value="UT"
              >
                Utah
              </option>
              <option
                label="Vermont"
                value="VT"
              >
                Vermont
              </option>
              <option
                label="Virginia"
                value="VA"
              >
                Virginia
              </option>
              <option
                label="Washington"
                value="WA"
              >
                Washington
              </option>
              <option
                label="West Virginia"
                value="WV"
              >
                West Virginia
              </option>
              <option
                label="Wisconsin"
                value="WI"
              >
                Wisconsin
              </option>
              <option
                label="Wyoming"
                value="WY"
              >
                Wyoming
              </option>
            </select>
            <span
              class="bp3-icon bp3-icon-double-caret-vertical"
              icon="double-caret-vertical"
            >
              <svg
                data-icon="double-caret-vertical"
                height="16"
                viewBox="0 0 16 16"
                width="16"
              >
                <desc>
                  double-caret-vertical
                </desc>
                <path
                  d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </div>
        </label>
      </div>
      <div
        class="sc-gzVnrw coBAaV"
      >
        <div
          class="sc-htoDjs GlcDy"
        >
          Click "Browse" to choose the appropriate file from your computer. This file should be a comma-separated list of all the jurisdictions participating in the audit, plus email addresses for audit administrators in each participating jurisdiction.
          <br />
          <br />
          <a
            href="/sample_jurisdiction_filesheet.csv"
            rel="noopener noreferrer"
            target="_blank"
          >
            (Click here to view a sample file in the correct format.)
          </a>
        </div>
      </div>
      <div
        class="sc-gzVnrw coBAaV"
      >
        <label
          class="bp3-file-input"
        >
          <input
            accept=".csv"
            name="csv"
            type="file"
          />
          <span
            class="bp3-file-upload-input"
          >
            Select a CSV...
          </span>
        </label>
      </div>
    </div>
    <div
      class="sc-htpNat gtULUV"
    >
      <button
        class="bp3-button bp3-intent-primary sc-EHOje gmpgQN"
        type="submit"
      >
        <span
          class="bp3-button-text"
        >
          Save & Next
        </span>
      </button>
    </div>
  </form>
</div>
`;

exports[`Audit Setup > Participants renders filled state correctly & resets file 1`] = `
<div>
  <form
    data-testid="form-one"
  >
    <div
      class="sc-bwzfXH jFbAHP"
    >
      <h2
        class="bp3-heading sc-bdVaJa ibVpPq"
      >
        Participants
      </h2>
      <div
        class="sc-gzVnrw coBAaV"
      >
        <label
          for="state"
        >
          Choose your state from the options below
          <br />
          <div
            class="bp3-html-select sc-iwsKbI iYAkEx"
          >
            <select
              data-testid="state-field"
              field="[object Object]"
              form="[object Object]"
              id="state"
            >
              <option
                value=""
              />
              <option
                label="Alabama"
                value="AL"
              >
                Alabama
              </option>
              <option
                label="Alaska"
                value="AK"
              >
                Alaska
              </option>
              <option
                label="Arizona"
                value="AZ"
              >
                Arizona
              </option>
              <option
                label="Arkansas"
                value="AR"
              >
                Arkansas
              </option>
              <option
                label="California"
                value="CA"
              >
                California
              </option>
              <option
                label="Colorado"
                value="CO"
              >
                Colorado
              </option>
              <option
                label="Connecticut"
                value="CT"
              >
                Connecticut
              </option>
              <option
                label="Delaware"
                value="DE"
              >
                Delaware
              </option>
              <option
                label="District Of Columbia"
                value="DC"
              >
                District Of Columbia
              </option>
              <option
                label="Florida"
                value="FL"
              >
                Florida
              </option>
              <option
                label="Georgia"
                value="GA"
              >
                Georgia
              </option>
              <option
                label="Hawaii"
                value="HI"
              >
                Hawaii
              </option>
              <option
                label="Idaho"
                value="ID"
              >
                Idaho
              </option>
              <option
                label="Illinois"
                value="IL"
              >
                Illinois
              </option>
              <option
                label="Indiana"
                value="IN"
              >
                Indiana
              </option>
              <option
                label="Iowa"
                value="IA"
              >
                Iowa
              </option>
              <option
                label="Kansas"
                value="KS"
              >
                Kansas
              </option>
              <option
                label="Kentucky"
                value="KY"
              >
                Kentucky
              </option>
              <option
                label="Louisiana"
                value="LA"
              >
                Louisiana
              </option>
              <option
                label="Maine"
                value="ME"
              >
                Maine
              </option>
              <option
                label="Maryland"
                value="MD"
              >
                Maryland
              </option>
              <option
                label="Massachusetts"
                value="MA"
              >
                Massachusetts
              </option>
              <option
                label="Michigan"
                value="MI"
              >
                Michigan
              </option>
              <option
                label="Minnesota"
                value="MN"
              >
                Minnesota
              </option>
              <option
                label="Mississippi"
                value="MS"
              >
                Mississippi
              </option>
              <option
                label="Missouri"
                value="MO"
              >
                Missouri
              </option>
              <option
                label="Montana"
                value="MT"
              >
                Montana
              </option>
              <option
                label="Nebraska"
                value="NE"
              >
                Nebraska
              </option>
              <option
                label="Nevada"
                value="NV"
              >
                Nevada
              </option>
              <option
                label="New Hampshire"
                value="NH"
              >
                New Hampshire
              </option>
              <option
                label="New Jersey"
                value="NJ"
              >
                New Jersey
              </option>
              <option
                label="New Mexico"
                value="NM"
              >
                New Mexico
              </option>
              <option
                label="New York"
                value="NY"
              >
                New York
              </option>
              <option
                label="North Carolina"
                value="NC"
              >
                North Carolina
              </option>
              <option
                label="North Dakota"
                value="ND"
              >
                North Dakota
              </option>
              <option
                label="Ohio"
                value="OH"
              >
                Ohio
              </option>
              <option
                label="Oklahoma"
                value="OK"
              >
                Oklahoma
              </option>
              <option
                label="Oregon"
                value="OR"
              >
                Oregon
              </option>
              <option
                label="Pennsylvania"
                value="PA"
              >
                Pennsylvania
              </option>
              <option
                label="Rhode Island"
                value="RI"
              >
                Rhode Island
              </option>
              <option
                label="South Carolina"
                value="SC"
              >
                South Carolina
              </option>
              <option
                label="South Dakota"
                value="SD"
              >
                South Dakota
              </option>
              <option
                label="Tennessee"
                value="TN"
              >
                Tennessee
              </option>
              <option
                label="Texas"
                value="TX"
              >
                Texas
              </option>
              <option
                label="Utah"
                value="UT"
              >
                Utah
              </option>
              <option
                label="Vermont"
                value="VT"
              >
                Vermont
              </option>
              <option
                label="Virginia"
                value="VA"
              >
                Virginia
              </option>
              <option
                label="Washington"
                value="WA"
              >
                Washington
              </option>
              <option
                label="West Virginia"
                value="WV"
              >
                West Virginia
              </option>
              <option
                label="Wisconsin"
                value="WI"
              >
                Wisconsin
              </option>
              <option
                label="Wyoming"
                value="WY"
              >
                Wyoming
              </option>
            </select>
            <span
              class="bp3-icon bp3-icon-double-caret-vertical"
              icon="double-caret-vertical"
            >
              <svg
                data-icon="double-caret-vertical"
                height="16"
                viewBox="0 0 16 16"
                width="16"
              >
                <desc>
                  double-caret-vertical
                </desc>
                <path
                  d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </div>
        </label>
      </div>
      <div
        class="sc-gzVnrw coBAaV"
      >
        <div
          class="sc-htoDjs GlcDy"
        >
          Click "Browse" to choose the appropriate file from your computer. This file should be a comma-separated list of all the jurisdictions participating in the audit, plus email addresses for audit administrators in each participating jurisdiction.
          <br />
          <br />
          <a
            href="/sample_jurisdiction_filesheet.csv"
            rel="noopener noreferrer"
            target="_blank"
          >
            (Click here to view a sample file in the correct format.)
          </a>
        </div>
      </div>
      <div
        class="sc-gzVnrw coBAaV"
      >
        <span>
          filename.csv
           
        </span>
        <button
          class="bp3-button sc-EHOje gmpgQN"
          type="button"
        >
          <span
            class="bp3-button-text"
          >
            Replace File
          </span>
        </button>
      </div>
    </div>
    <div
      class="sc-htpNat gtULUV"
    >
      <button
        class="bp3-button bp3-intent-primary sc-EHOje gmpgQN"
        type="submit"
      >
        <span
          class="bp3-button-text"
        >
          Save & Next
        </span>
      </button>
    </div>
  </form>
</div>
`;

exports[`Audit Setup > Participants renders filled state correctly & resets file 2`] = `
<div>
  <form
    data-testid="form-one"
  >
    <div
      class="sc-bwzfXH jFbAHP"
    >
      <h2
        class="bp3-heading sc-bdVaJa ibVpPq"
      >
        Participants
      </h2>
      <div
        class="sc-gzVnrw coBAaV"
      >
        <label
          for="state"
        >
          Choose your state from the options below
          <br />
          <div
            class="bp3-html-select sc-iwsKbI iYAkEx"
          >
            <select
              data-testid="state-field"
              field="[object Object]"
              form="[object Object]"
              id="state"
            >
              <option
                value=""
              />
              <option
                label="Alabama"
                value="AL"
              >
                Alabama
              </option>
              <option
                label="Alaska"
                value="AK"
              >
                Alaska
              </option>
              <option
                label="Arizona"
                value="AZ"
              >
                Arizona
              </option>
              <option
                label="Arkansas"
                value="AR"
              >
                Arkansas
              </option>
              <option
                label="California"
                value="CA"
              >
                California
              </option>
              <option
                label="Colorado"
                value="CO"
              >
                Colorado
              </option>
              <option
                label="Connecticut"
                value="CT"
              >
                Connecticut
              </option>
              <option
                label="Delaware"
                value="DE"
              >
                Delaware
              </option>
              <option
                label="District Of Columbia"
                value="DC"
              >
                District Of Columbia
              </option>
              <option
                label="Florida"
                value="FL"
              >
                Florida
              </option>
              <option
                label="Georgia"
                value="GA"
              >
                Georgia
              </option>
              <option
                label="Hawaii"
                value="HI"
              >
                Hawaii
              </option>
              <option
                label="Idaho"
                value="ID"
              >
                Idaho
              </option>
              <option
                label="Illinois"
                value="IL"
              >
                Illinois
              </option>
              <option
                label="Indiana"
                value="IN"
              >
                Indiana
              </option>
              <option
                label="Iowa"
                value="IA"
              >
                Iowa
              </option>
              <option
                label="Kansas"
                value="KS"
              >
                Kansas
              </option>
              <option
                label="Kentucky"
                value="KY"
              >
                Kentucky
              </option>
              <option
                label="Louisiana"
                value="LA"
              >
                Louisiana
              </option>
              <option
                label="Maine"
                value="ME"
              >
                Maine
              </option>
              <option
                label="Maryland"
                value="MD"
              >
                Maryland
              </option>
              <option
                label="Massachusetts"
                value="MA"
              >
                Massachusetts
              </option>
              <option
                label="Michigan"
                value="MI"
              >
                Michigan
              </option>
              <option
                label="Minnesota"
                value="MN"
              >
                Minnesota
              </option>
              <option
                label="Mississippi"
                value="MS"
              >
                Mississippi
              </option>
              <option
                label="Missouri"
                value="MO"
              >
                Missouri
              </option>
              <option
                label="Montana"
                value="MT"
              >
                Montana
              </option>
              <option
                label="Nebraska"
                value="NE"
              >
                Nebraska
              </option>
              <option
                label="Nevada"
                value="NV"
              >
                Nevada
              </option>
              <option
                label="New Hampshire"
                value="NH"
              >
                New Hampshire
              </option>
              <option
                label="New Jersey"
                value="NJ"
              >
                New Jersey
              </option>
              <option
                label="New Mexico"
                value="NM"
              >
                New Mexico
              </option>
              <option
                label="New York"
                value="NY"
              >
                New York
              </option>
              <option
                label="North Carolina"
                value="NC"
              >
                North Carolina
              </option>
              <option
                label="North Dakota"
                value="ND"
              >
                North Dakota
              </option>
              <option
                label="Ohio"
                value="OH"
              >
                Ohio
              </option>
              <option
                label="Oklahoma"
                value="OK"
              >
                Oklahoma
              </option>
              <option
                label="Oregon"
                value="OR"
              >
                Oregon
              </option>
              <option
                label="Pennsylvania"
                value="PA"
              >
                Pennsylvania
              </option>
              <option
                label="Rhode Island"
                value="RI"
              >
                Rhode Island
              </option>
              <option
                label="South Carolina"
                value="SC"
              >
                South Carolina
              </option>
              <option
                label="South Dakota"
                value="SD"
              >
                South Dakota
              </option>
              <option
                label="Tennessee"
                value="TN"
              >
                Tennessee
              </option>
              <option
                label="Texas"
                value="TX"
              >
                Texas
              </option>
              <option
                label="Utah"
                value="UT"
              >
                Utah
              </option>
              <option
                label="Vermont"
                value="VT"
              >
                Vermont
              </option>
              <option
                label="Virginia"
                value="VA"
              >
                Virginia
              </option>
              <option
                label="Washington"
                value="WA"
              >
                Washington
              </option>
              <option
                label="West Virginia"
                value="WV"
              >
                West Virginia
              </option>
              <option
                label="Wisconsin"
                value="WI"
              >
                Wisconsin
              </option>
              <option
                label="Wyoming"
                value="WY"
              >
                Wyoming
              </option>
            </select>
            <span
              class="bp3-icon bp3-icon-double-caret-vertical"
              icon="double-caret-vertical"
            >
              <svg
                data-icon="double-caret-vertical"
                height="16"
                viewBox="0 0 16 16"
                width="16"
              >
                <desc>
                  double-caret-vertical
                </desc>
                <path
                  d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                  fill-rule="evenodd"
                />
              </svg>
            </span>
          </div>
        </label>
      </div>
      <div
        class="sc-gzVnrw coBAaV"
      >
        <div
          class="sc-htoDjs GlcDy"
        >
          Click "Browse" to choose the appropriate file from your computer. This file should be a comma-separated list of all the jurisdictions participating in the audit, plus email addresses for audit administrators in each participating jurisdiction.
          <br />
          <br />
          <a
            href="/sample_jurisdiction_filesheet.csv"
            rel="noopener noreferrer"
            target="_blank"
          >
            (Click here to view a sample file in the correct format.)
          </a>
        </div>
      </div>
      <div
        class="sc-gzVnrw coBAaV"
      >
        <label
          class="bp3-file-input"
        >
          <input
            accept=".csv"
            name="csv"
            type="file"
          />
          <span
            class="bp3-file-upload-input"
          >
            Select a CSV...
          </span>
        </label>
      </div>
    </div>
    <div
      class="sc-htpNat gtULUV"
    >
      <button
        class="bp3-button bp3-intent-primary sc-EHOje gmpgQN"
        type="submit"
      >
        <span
          class="bp3-button-text"
        >
          Save & Next
        </span>
      </button>
    </div>
  </form>
</div>
`;
